File: Data Structures & Algorithms/duplicate-integer/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const tmp={};
        for(let i of nums){
            if(tmp[i]){
                return true;
            }
            tmp[i]=true;
        }
        return false;
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        for(int i=0;i<n;i++){
            int k=nums[i]%(n+1);
            if(nums[k-1]/(n+1)){
                return k;
            }
            nums[k-1]+=n+1;
        }
        return -1;
    }
};
